DigitalLearningSolutions.Web/Scripts/trackingSystem/reports.ts

```diff
@@ -9,6 +9,8 @@ const path = getPathForEndpoint('TrackingSystem/Centre/Reports/Data');
 const activityToggleableRowDisplayNone = 'none';
 const activityToggleableRowDisplayTableRow = 'table-row';
 
+const maxNumberOfEntriesForActivityGraph = 17;

```

Comment:
17 feels a little bit restrictive, and limits what you can do with a couple of the intervals - e.g. you can't do daily over a month, or weekly over a half-year. I'd hazard a guess that people will generally be using a tablet or computer to look at reports, so they'd probably find it a bit annoying to be restricted by the size of a phone screen.
E.g. IPad view of 6 months weekly: 
![image](https://user-images.githubusercontent.com/34240660/140779870-551567f0-99d0-439c-9aa5-e4ff92d5dfc6.png)

I reckon this is a question worth asking HEE about.

I'd be inclined to raise this to 30 or so and just ignore phones, or maybe set up breakpoints to chose the limit based on the width of the element, but responding to screen dimention changes gets fiddly e.g. when you rotate a phone screen.

Reply:
The problem is typically at the bottom of the desktop range on these pages with a sidebar. 30 gets a bit congested but is probably just about manageable:
![image](https://user-images.githubusercontent.com/59561751/140781601-903bf073-894d-4627-bdfe-de28ef813d5d.png)

Reply:
Also I could implement a different limit for if the page is loaded on mobile initially. I think it is too much trouble to change responsively if the window is resized, other than perhaps dropping the graph if they go from tablet to mobile with 30 data points. Thoughts?

Reply:
Yeah, sounds reasonable to determine the limit on page load. Extra good if we can make that clear on the alternative text. Dropping the graph if they go from tablet to mobile - that sounds nearly as hard as recalculating on every window resize, no?

Reply:
Yeah it's still recalculating on window resize, but at least it isn't recalculating a limit and having to repeatedly call the server to get the data again.

Reply:
I think we should still show a graph on mobile, so still have that 17-point limit or whatever.
You could just get the js to store the data rather than re-requesting it.